src/python_testing/TC_SC_3_6.py

```diff
@@ -0,0 +1,226 @@
+
+        asserts.assert_equal(len(client_list), num_fabrics_to_commission *
+                             num_controllers_per_fabric, "Must have the right number of clients")
+
+        # Before subscribing, set the NodeBabel to "Before Subscriptions"
+        logging.info("Pre-conditions: writing initial value of NodeLabel, so that we can control for change of attribute detection")
+        await client_list[0].WriteAttribute(self.dut_node_id, [(0, Clusters.Basic.Attributes.NodeLabel(value=BEFORE_LABEL))])
+
+        # Subscribe with all clients to NodeLabel attribute
+        subscriptions = []
+        sub_handlers = []
+        resub_catchers = []
+
+        logging.info("Step 1 (first part): Establish subscription with all 15 clients")
+        for client in client_list:
+            logging.info("Establishing subscription from controller node %s" % client.name)
+            sub = await client.ReadAttribute(nodeid=self.dut_node_id, attributes=[(0, Clusters.Basic.Attributes.NodeLabel)],
+                                             reportInterval=(min_report_interval_sec, max_report_interval_sec), keepSubscriptions=False)
+            subscriptions.append(sub)
+
+            attribute_handler = AttributeChangeAccumulator(name=client.name)
+            sub.SetAttributeUpdateCallback(attribute_handler)
+            sub_handlers.append(attribute_handler)
+
+            resub_catcher = ResubscriptionCatcher(name=client.name)
+            sub.SetResubscriptionAttemptedCallback(resub_catcher, isAsync=True)
+            resub_catchers.append(resub_catcher)
+
+            if sub_liveness_override_ms is not None:
+                logging.warning("Overriding subscription liveness to check %dms! NOT FOR CERTIFICATION!" % sub_liveness_override_ms)
+                sub.OverrideLivenessTimeoutMs(sub_liveness_override_ms)
+
+        asserts.assert_equal(len(subscriptions), num_fabrics_to_commission *
+                             num_controllers_per_fabric, "Must have the right number of subscriptions")
+
+        # Trigger a change on NodeLabel
+        logging.info("Step 1 (second part): Change attribute with one client, and validate all clients observe the change on same session")
+        await asyncio.sleep(1)
+        await client_list[0].WriteAttribute(self.dut_node_id, [(0, Clusters.Basic.Attributes.NodeLabel(value=AFTER_LABEL))])
+
+        # Await a stabilization delay in increments to let the coroutines run
+        start_time = time.time()
+        while (time.time() - start_time) < stabilization_delay_sec:
+            await asyncio.sleep(0.05)
+
+        # After stabilization, validate no resubscriptions and all nodes have seen an update
+
+        logging.info("Validation of results")
+        # First check: all subs seeing update
+        failed = False
+        for handler in sub_handlers:
+            data_update_count = 0
+            for item in handler.all_data:
+                if item['value'] == AFTER_LABEL:
```

Comment:
The `AttributeChangeAccumulator` is not actually needed - you can directly call `SubscriptionTransaction.GetAttribute` directly instead.

To avoid the race condition, you should shutdown all the subs before doing this check (you need to shutdown the subs anyways to avoid messages being dispatched onto a now-defunct event loop).

Reply:
The subs cannot yet be shutdown with existing APIs. We need the "no valid subscriptions and `keepSubcriptions == False`" change to turn down the subs before that can be done.

Overall, the goal is to ensure that *all the CASE sessions remained alive*. Maybe I can skip trying to look at the value given the liveness check elsewhere.